rotkehlchen/db/addressbook.py

```diff
@@ -42,20 +47,36 @@ def write_ctx(self, book_type: AddressbookType) -> Generator['DBCursor', None, N
     def get_addressbook_entries(  # pylint: disable=no-self-use
             self,
             cursor: 'DBCursor',
-            addresses: Optional[list[ChecksumEvmAddress]] = None,
+            addresses: Optional[list[
+                tuple[ChecksumEvmAddress, Optional[SupportedBlockchain]]
+            ]] = None,
     ) -> list[AddressbookEntry]:
+        entries = []
         if addresses is None:
-            cursor.execute('SELECT address, name FROM address_book')
+            cursor.execute('SELECT address, name, blockchain FROM address_book')
+            for address, name, blockchain_str in cursor:
+                deserialized_blockchain = SupportedBlockchain(blockchain_str)
+                entries.append(AddressbookEntry(
+                    address=ChecksumEvmAddress(address),
+                    name=name,
+                    blockchain=deserialized_blockchain,
+                ))
         else:
-            questionmarks = ','.join('?' * len(addresses))
-            cursor.execute(
-                f'SELECT address, name FROM address_book WHERE address IN ({questionmarks})',
-                addresses,
-            )
-
-        entries = []
-        for address, name in cursor:
-            entries.append(AddressbookEntry(address=ChecksumEvmAddress(address), name=name))
+            for address, blockchain in addresses:
+                query = 'SELECT address, name, blockchain FROM address_book WHERE address = ?'
+                bindings = [address]
```

Comment:
this is ugly, you are doing one query per item in the tuple. You could build the queries and do a `executemany`. It would be even less queries if you group by blockchain and then do as it was before but grouped by blockechain. You would have as many as `len(distinct_blockchains)` queries